notebooks/Data_Attribute_Recommendation_Regression_Model_Template.ipynb

```diff
@@ -0,0 +1,1933 @@
KJHj+5huiDrm2Kn0gSz2cacOXND+fLKCz3rw2wDC+P3oyPgCDgCjoAj4Ag4Ao5Ax0bAHaUOVD+27idW6TvXfEvmzZ8vJ732NLnqC18OGye85pSTA8k7zztXnnrmmTBtjI0VcJzq6ppPSTv7rLfK3Llz5bgTT9GNEL4Y1iht3ly3I5so4w9+4GJds3SInH3O+XLCKafKAw8+JD3UKcgKEWsgYaSHqW7Xfvsb8uvf/k5OOfUNwi51V3/r6zJmdP2IGdues6PcEbo2inDIIQeF6+OOfVW4jv/gWF2qa7R+ps4WU+/uu/+BsH5oi24aQdpZbzkjbHjBdMLXnHJSyPObV18biwijUPPmzpPXnvamIIMNLM4/5x0yYsRw+e6135a/3H2PvPqk18q733uh9NJ1SewSyNojnNQ3nXGWrtv6SxN5XLxJN3vAAXrzmWfllpcpgj9SR/exKfXrspoJ8ghHwBFwBBwBR8ARcAQcgQ6JQJfJkyc3Di3w9r4loRz+NJ60ONMrKy2Oj8/hK/c6iy8ZP1PX9UyZMsVUbJUjH55ltAXnIA4Y9bW1m3U0pelITExj5+vWrZO+ffvaZe6R3euYMtenT59cukKJpeSZJ4v1Uzh3aeUEG0aTcNIYWWJnPX7JAH9dXXqZ2IadzRuYfhgH4vspZmlT/ciLEI+CpZUX3ZNy4zz83BFwBBwBR8ARcAQcgSwEjjjiCJmga73jYHaJHS3Nru2YjE+7TtJCE8fF58afpCkmPo8n5rfzrHwtPe/YEt5YbnNrMk718w6DQNaHZ3Gc0pyHNMWLdZLgZXMJfi0NpeSZlxeORq9eTZ0Yo4+xiZ0WS7cj0+n4pQXbjCKZlhUPXVpeaeV1JymJql87Ao6AI+AIOAKOgCPQ8RFoOjzR8fV1DR0BR8ARcAQcAUfAEXAEHAFHwBFodQTcUWp1iD0DR8ARcAQcAUfAEXAEHAFHwBHobAi4o9TZasz17dQIrFy3Rf75/JpQBpYE3vfcGlm1YWunLlOW8k/P3iC3P7oiJD87Z4PMWFybRerxrYzA49PXy/wVm0vKJa6/khid2BFwBBwBR8AR2EkQ8DVKHaAit2zdLlfftVD+9PgqWa6G9O7Dq+U9Jw2Ts44eXFC7KWoA7Te2l/Subu7zvvO70+X0QwfKO44d0ihn+dotcsY3psnNH9hd9hhVeAMIY9y8ZbucctULdinV3bvIAeN6y8WvGyF7ji5eTqMAPdlQu02mztsoh09s2YYRscxSzl9ZtEn6VHeTUYPS1y3lydqq25P/4G+L5R/PrhHkgMVbjhpcsM5wGP73BzNk+vcOlm3qKV14wwy59cOT5Kg9mm6ykcR73LAecuiEPnLOcUNk9KAeeaqFtMWr62TV+q2yVxl1o0WTN3/tJele1UXu+PieTfKK21uh+vvVQ8vl3mdWB1w+95t5cojq/7mzxjSRV+iiUB5Z/En8KMu+ep9c+JrhctD43llsO238pTfPksknD5P/PWV4bhnj+o3rL5epxMTpizfJe743Q1574AD53Nt2tIcf/n2J/OLBZY3SdhvaQ07Yt7+8T+uM3TX/+PhK+ZHS3HVF/QeyGwn9xBFwBBwBR8ARaCUEmlvXrZSRi81G4OdqHGAknH3MYLnm3eODQfl/v5gTnIhsLhGM9Xdc87LMXJL+pv6lBRtl6ZpoG3AVVrtlmyxYuVk21W3LE90sbYvmBd+xe/cLxubJ+/eXx15ZJ29Sg3qJGuXlhP/OWC/nfeeVclgrwvPBH8+SO6fUj3iUKhDea+5apI5ILzX+x8r4YdVCnX3vnkWlikqlj/HGae7WtYt8/6+L5dQvvSgvL6zfbS+VsSHyx/9YIlfdNi+PJDPtgalr5IX5G+UZHRF69OV1jXTJ9lao/r5y7m7y7y/t28hfzkmhPLJkxvjhHL3hkIHhfnrLN6fJ7KXp90uWrF0lPlm/lai/NOx++eDy0Gf88l/LZOPmHf3QQu1fuKa+3n3iUH2J0VW+cecC+eLv5gcxy9ZskRkZfV1aPh7nCDgCjoAj4Ai0FAEfUWopghXgx+HYZ0wvufS0kUHaqQcPkMveOFIG962vHqZmffrWufLvl9bKpJE9lW5EeNP69mvqnYwLvj89vLX/5BmjS9KGqV+nfOEFOVdHKX77nxWyQkezXqNveb/4Dv24azd9hZsSXq+6nbhf/5BywUnD5bjPPi8Pvbg25M9UnW/9caE8NWu97DakWj7z1tFyzF79wsjR534zV+5/fm0YiWKU69zjhsplP50tGLQnXjlVacfIPU+vlokjqoO85+dulNcfPDAYTP/38zmyVEfCGL161wlDQ95/0rfLN923VKdzbZIDdYTganUwRwzoLrc+tEyenLlB+vXsFt5AM9r18TePCs5nXJyv3bEgOBzXq/Nxv06F+/Vle4Tyf/638+Q/L62TXj26hrzT3sAzSvI31fV7k2vkNDXACW971WA5Zs++0rdhZz4cja/cvkAYQWIE6Ep9c54cNYr1yTo3vBkNYHSF0cCP/Wy2/PGTewVH+Tt/WSR3/XeVrNE2Qt19+ZyxOoqzRn7xr+WyVUcqwfa2j+0hyxS/YvX5+QPL5I2HDZSVOiIFnqZ33N5O1bq5+4lVTevvqVUytH/3gM1rDugvNToy+si0dXLde2tC8Zap037eda8Idbvfbr3kq+pIjRtaLbdofrQZ6pCAc4ZxfNPFuzdrI6ccMEC+/ocFcqfWP4Y0o6VpdRQE6R/Dj+v/VQP8sE88G6Y+vufEYWHKY9p9BS3O6JXaFqi/8arjp7V9vkrrl3skq428Ru+lyScPl589sDTQfeT0kWHEkntiQO9ujTKsjTIKfOeUlbKnjuxSdkbdcE65/7/w9rHBOc669xmJ/chPZstFrxsu12kb6KqO9KWvHyFn6qgm4WHF/Vt/XCDTdbrjSXq/4gRZyGqbcf0yOjpOR3Ss/ugrwP3PWuc4M7Q1RoN6631yud6fk/S+ffSV9XrvrRdeovyf9kXcj8mAHrc9s
```

Comment:
I generally like to see the outputs of `read_model_collection`, `read_job_collection` etc., and perhaps we can move those bits to the places right after the respective things are created? Also, a bit of "explanation text" could be helpful

Reply:
Interesting point! For last year's workshop, I had opted to keep the `read_*_collection` methods as "optional exercises" towards the end of the notebook. They are also in this notebook towards the end.

I did not have the various methods in the main flow because I wanted to focus a bit on the story and not so much on all the details. I think that is good for a workshop. Maybe for self-study offline outside of a workshop, it would be OK to have more details at each stage.

I am fine with either way, but if we want to use the notebooks also for a workshop, I would prefer keeping the things optional. Or we can make a single example at the beginning for the dataset schema for `get_dataset_schema_by_id` and `read_dataset_schemas_collection` and state that there are similar methods for the other resources.

Reply:
I agree that this might be useful but seldom do we encounter someone reading all the text in the notebook. Most users simply run the notebooks to see the outputs and get to their results as quickly as possible. We can keep this as a way forward to improve the notebooks. I've added an issue linking to this comment so that we have the context in the future: https://github.com/SAP/data-attribute-recommendation-python-sdk/issues/119